src/blk/BlockDevice.cc

```diff
@@ -90,67 +86,63 @@ void IOContext::release_running_aios()
 #endif
 }
 
-BlockDevice *BlockDevice::create(CephContext* cct, const string& path,
-				 aio_callback_t cb, void *cbpriv, aio_callback_t d_cb, void *d_cbpriv)
-{
-  string type = "kernel";
-  char buf[PATH_MAX + 1];
-  int r = ::readlink(path.c_str(), buf, sizeof(buf) - 1);
-  if (r >= 0) {
-    buf[r] = '\0';
-    char *bname = ::basename(buf);
-    if (strncmp(bname, SPDK_PREFIX, sizeof(SPDK_PREFIX)-1) == 0)
-      type = "ust-nvme";
-  }
+BlockDevice::block_device_t
+BlockDevice::detect_device_type(const std::string& path) {
```

Comment:
could you move the opening "{" to a new line?

Reply:
done